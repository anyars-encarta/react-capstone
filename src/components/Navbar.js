import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faMicrophoneAlt, faUserCog, faChevronLeft } from '@fortawesome/free-solid-svg-icons';
import { useHistory } from 'react-router-dom';
import '../styles/Navbar.css';

const Navbar = () => {
  const history = useHistory();

  const handleArrowLeftClick = () => {
    history.push('/');
  };

  const handleArrowLeftKeyDown = (event) => {
    if (event.key === 'Enter' || event.key === ' ') {
      history.push('/');
    }
  };

  return (
    <div className="nav-container">
      <nav className="navbar">
        <div
          className="left"
          onClick={handleArrowLeftClick}
          onKeyDown={handleArrowLeftKeyDown}
          role="button"
          tabIndex="0"
        >
          <FontAwesomeIcon icon={faChevronLeft} className="arrow-left" />
          <p className="case-year">2023</p>
        </div>

        <p className="continent-name">WORLD COVID CASES</p>

        <div className="right">
          <FontAwesomeIcon icon={faMicrophoneAlt} />
          <FontAwesomeIcon icon={faUserCog} />
        </div>
      </nav>
    </div>
  );
};

export default Navbar;
