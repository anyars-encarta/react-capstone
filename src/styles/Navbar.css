/* Navbar.css */
.nav-container {
  background-color: #e14984;
  color: #ffff;
  position: fixed;
  width: 100%;
  top: 0;
  z-index: 100;
}

.navbar {
  display: flex;
  justify-content: space-between;
  padding: 10px;
}

.left {
  display: flex;
  align-items: center;
  gap: 10px;
}

.arrow-left {
  cursor: pointer;
}

.right {
  display: flex;
  align-items: center;
  gap: 10px;
}

.continent-name {
  font-weight: bold;
}
