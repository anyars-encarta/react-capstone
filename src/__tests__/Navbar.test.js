import React from 'react';
import { render } from '@testing-library/react';
import Navbar from '../components/Navbar';

test('Navbar renders with correct content', () => {
  const { asFragment, getByText } = render(<Navbar />);

  // Assert that the component renders with the correct text content
  expect(getByText('2023')).toBeInTheDocument();
  expect(getByText('WORLD COVID CASES')).toBeInTheDocument();

  // Create a snapshot of the component
  expect(asFragment()).toMatchSnapshot();
});
