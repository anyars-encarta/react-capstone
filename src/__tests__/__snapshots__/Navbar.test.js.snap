// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders with correct content 1`] = `
<DocumentFragment>
  <div
    class="nav-container"
  >
    <nav
      class="navbar"
    >
      <div
        class="left"
        role="button"
        tabindex="0"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-chevron-left arrow-left"
          data-icon="chevron-left"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l192 192c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L77.3 256 246.6 86.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-192 192z"
            fill="currentColor"
          />
        </svg>
        <p
          class="case-year"
        >
          2023
        </p>
      </div>
      <p
        class="continent-name"
      >
        WORLD COVID CASES
      </p>
      <div
        class="right"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-microphone-lines "
          data-icon="microphone-lines"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M96 96V256c0 53 43 96 96 96s96-43 96-96H208c-8.8 0-16-7.2-16-16s7.2-16 16-16h80V192H208c-8.8 0-16-7.2-16-16s7.2-16 16-16h80V128H208c-8.8 0-16-7.2-16-16s7.2-16 16-16h80c0-53-43-96-96-96S96 43 96 96zM320 240v16c0 70.7-57.3 128-128 128s-128-57.3-128-128V216c0-13.3-10.7-24-24-24s-24 10.7-24 24v40c0 89.1 66.2 162.7 152 174.4V464H120c-13.3 0-24 10.7-24 24s10.7 24 24 24h72 72c13.3 0 24-10.7 24-24s-10.7-24-24-24H216V430.4c85.8-11.7 152-85.3 152-174.4V216c0-13.3-10.7-24-24-24s-24 10.7-24 24v24z"
            fill="currentColor"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-user-gear "
          data-icon="user-gear"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224 0a128 128 0 1 1 0 256A128 128 0 1 1 224 0zM178.3 304h91.4c11.8 0 23.4 1.2 34.5 3.3c-2.1 18.5 7.4 35.6 21.8 44.8c-16.6 10.6-26.7 31.6-20 53.3c4 12.9 9.4 25.5 16.4 37.6s15.2 23.1 24.4 33c15.7 16.9 39.6 18.4 57.2 8.7v.9c0 9.2 2.7 18.5 7.9 26.3H29.7C13.3 512 0 498.7 0 482.3C0 383.8 79.8 304 178.3 304zM436 218.2c0-7 4.5-13.3 11.3-14.8c10.5-2.4 21.5-3.7 32.7-3.7s22.2 1.3 32.7 3.7c6.8 1.5 11.3 7.8 11.3 14.8v30.6c7.9 3.4 15.4 7.7 22.3 12.8l24.9-14.3c6.1-3.5 13.7-2.7 18.5 2.4c7.6 8.1 14.3 17.2 20.1 27.2s10.3 20.4 13.5 31c2.1 6.7-1.1 13.7-7.2 17.2l-25 14.4c.4 4 .7 8.1 .7 12.3s-.2 8.2-.7 12.3l25 14.4c6.1 3.5 9.2 10.5 7.2 17.2c-3.3 10.6-7.8 21-13.5 31s-12.5 19.1-20.1 27.2c-4.8 5.1-12.5 5.9-18.5 2.4l-24.9-14.3c-6.9 5.1-14.3 9.4-22.3 12.8l0 30.6c0 7-4.5 13.3-11.3 14.8c-10.5 2.4-21.5 3.7-32.7 3.7s-22.2-1.3-32.7-3.7c-6.8-1.5-11.3-7.8-11.3-14.8V454.8c-8-3.4-15.6-7.7-22.5-12.9l-24.7 14.3c-6.1 3.5-13.7 2.7-18.5-2.4c-7.6-8.1-14.3-17.2-20.1-27.2s-10.3-20.4-13.5-31c-2.1-6.7 1.1-13.7 7.2-17.2l24.8-14.3c-.4-4.1-.7-8.2-.7-12.4s.2-8.3 .7-12.4L343.8 325c-6.1-3.5-9.2-10.5-7.2-17.2c3.3-10.6 7.7-21 13.5-31s12.5-19.1 20.1-27.2c4.8-5.1 12.4-5.9 18.5-2.4l24.8 14.3c6.9-5.1 14.5-9.4 22.5-12.9V218.2zm92.1 133.5a48.1 48.1 0 1 0 -96.1 0 48.1 48.1 0 1 0 96.1 0z"
            fill="currentColor"
          />
        </svg>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
